File: webclient/src/app/core/model/seasonteamid.ts
export class SeasonTeamId {
 private _seasonId: string;
 private _teamId: string;

  constructor(seasonId: string, teamId: string) {
    this._seasonId = seasonId;
    this._teamId = teamId;
  }

  get seasonId(): string {
    return this._seasonId;
  }

  set seasonId(value: string) {
    this._seasonId = value;
  }

  get teamId(): string {
    return this._teamId;
  }

  set teamId(value: string) {
    this._teamId = value;
  }
}

File: core/VolleyballApp/src/main/java/com/engineeringthesis/VolleyballApp/data/model/SeasonTeamId.java
package com.engineeringthesis.VolleyballApp.data.model;

import jakarta.persistence.Embeddable;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.io.Serializable;
import java.util.Objects;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Embeddable
public class SeasonTeamId implements Serializable {

    private String seasonId;
    private String teamId;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        SeasonTeamId that = (SeasonTeamId) o;
        return seasonId.equals(that.seasonId) && teamId.equals(that.teamId);
    }

    @Override
    public int hashCode() {
        return Objects.hash(seasonId, teamId);
    }
}
